### PR TITLE
Component for NFT Img/Mp4 -> use it in `NftTierCard` and `ConfirmPayModal`

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

- A frustrating hydra-killing effort but finally got a  `JuiceVideoThumbnailOrImage` component working across the `NftTierCard` and pay modal.
- We should be able to apply this in a few more existing and new place (not necessary for this PR)

## Screenshots or screen recordings

https://user-images.githubusercontent.com/96150256/218666869-d7b496cf-fad9-4768-ac15-61302241e1f1.mp4

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/utils/nftRewards.ts [hunk 1]
```diff
@@ -388,12 +388,14 @@ export function sumTierFloors(
   rewardTiers: NftRewardTier[],
   tierIds?: number[],
 ) {
-  if (!tierIds) return 0
+  if (tierIds !== undefined && tierIds.length === 0) return 0
```
Comment (johnnyd-eth) [inline src/utils/nftRewards.ts:391]: Unrelated fix to this: 

Screen Shot 2023-02-14 at 5 27 53 pm

File: src/components/v2v3/shared/NftVideo/JuiceVideoThumbnailOrImage.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+import { JuiceVideoThumbnail, PlayIconPosition } from './JuiceVideoThumbnail'
+
+export function JuiceVideoThumbnailOrImage({
+  isSelected,
+  playIconPosition,
+  ...props
+}: ImageProps & {
+  isSelected?: boolean
+  playIconPosition?: PlayIconPosition
+  src: string
```
Comment (tomquirk) [inline src/components/v2v3/shared/NftVideo/JuiceVideoThumbnailOrImage.tsx:16]: **question** is `src` not already in ImageProps?

File: src/components/NftRewards/NftTierCard.tsx [hunk 3]
```diff
@@ -13,6 +10,8 @@ import { ipfsToHttps } from 'utils/ipfs'
 import { NftPreview } from './NftPreview'
 import { QuantitySelector } from './QuantitySelector'
 
+const NFT_DISPLAY_HEIGHT = '141px'
```
Comment (tomquirk) [inline src/components/NftRewards/NftTierCard.tsx:13]: this feels overly specific. is there an existing tailwind class we can use for this? perhaps `h-32` or `h-36`?

Comment (tomquirk) [reply src/components/NftRewards/NftTierCard.tsx:13]: https://tailwindcss.com/docs/height

File: src/components/v2v3/V2V3Project/V2V3PayButton/V2V3ConfirmPayModal/NftRewardCell.tsx [hunk 4]
```diff
@@ -19,19 +20,28 @@ export function NftRewardCell({
...
-        const tier = nftRewards.find(tier => tier.id === parseInt(tierId))
+        const tier = nftRewards.find(_tier => _tier.id === parseInt(tierId))
         if (!tier?.id) return
         const tierCount = uniqueTiersIdsAndCounts[tier.id]
         const isLink = tier.externalLink
 
         return (
           <div className="flex items-center justify-between" key={idx}>
-            <div className="flex items-center justify-start">
+            <div className="flex h-[50px] items-center justify-start">
```
Comment (tomquirk) [inline src/components/v2v3/V2V3Project/V2V3PayButton/V2V3ConfirmPayModal/NftRewardCell.tsx:30]: Same here, perhaps `h-12` or `h-14` are possibilities.

**Review by tomquirk — CHANGES_REQUESTED** [review]
Awesome refactor here @johnnyd-eth , coupla comments

File: src/components/v2v3/shared/NftVideo/JuiceVideoThumbnailOrImage.tsx (hunk 2 above)
Comment (johnnyd-eth) [reply src/components/v2v3/shared/NftVideo/JuiceVideoThumbnailOrImage.tsx:16]: Can be undefined in ImageProps is the only reason I included here

File: src/components/NftRewards/NftTierCard.tsx (hunk 3 above)
Comment (johnnyd-eth) [reply src/components/NftRewards/NftTierCard.tsx:13]: The reason I was so specific is because that happens to be the width that results from the 3-col `Row` (of NFT cards) with a gutter of `24px`. I've reduced that gutter a bit so that the cards are now 144px wide instead of 141px, which means we can just use `h-36` to set the widths of the image/videos :)

File: src/components/v2v3/V2V3Project/V2V3PayButton/V2V3ConfirmPayModal/NftRewardCell.tsx (hunk 4 above)
Comment (johnnyd-eth) [reply src/components/v2v3/V2V3Project/V2V3PayButton/V2V3ConfirmPayModal/NftRewardCell.tsx:30]: Done
